packages/blade/src/components/Menu/_decisions/decisions.md

```diff
@@ -0,0 +1,372 @@
+
+Since our usecases of Menu were simple and small, we added triggers like DropdownButton, DropdownLink to also handle some of the menu cases inside our existing Dropdown.
+
+But as we move into the next set of Menu usecases, we see the complexity of Menu increasing exponentially. Adding these usecases to Dropdown would mean- bloating dropdown with more unrelated code and bundle-size, ending up with inifinite edge cases, complex keyboard navigations for consumers, and bunch of if-else conditions internally to handle 2 very different usecases.
+
+#### 3. Two Different Usecases
+
+Normally you would see design-systems having 2 components. 1 for SelectInput, AutoComplete, etc and 2nd for Menu, ContextMenu, ActionMenu, etc
+
+Examples
+
+- [Primer SelectPanel](https://primer.style/components/selectpanel/react/alpha), [Primer ActionMenu](https://primer.style/components/action-menu/react/beta)
+- [Ariakit Select](https://ariakit.org/components/select), [Ariakit Combobox](https://ariakit.org/components/combobox), [Ariakit Menu](https://ariakit.org/components/menu)
+
+  > [!Note]
+  >
+  > AriaKit also has Select vs Menu usecases mentioned in their [Select vs Menu documentation](https://ariakit.org/components/menu#should-i-use-menu-or-select). You can check it out to understand their perspective
+
+- [Radix Dropdown Menu](https://www.radix-ui.com/themes/docs/components/dropdown-menu), [Radix Context Menu](https://www.radix-ui.com/themes/docs/components/context-menu), [Radix Select](https://www.radix-ui.com/themes/docs/components/select)
+
+Going forward, our Dropdown component will continue to cover SelectInput, AutoComplete, etc usecases where selections are happening.
+
+And Menu will cover usecases which trigger certain actions, has complex interactive / non-interactive things inside the Overlay, requires submenus, custom triggers, etc.
+
+#### 4. Impossible Scenarios
+
+Putting menu usecases inside Dropdown also leads to more impossible scenarios in API. E.g-
+
+- What does selection in submenu mean? Submenu items shouldn't be selectable
+- What do custom components inside DropdownOverlay do? they can't be selected so shouldn't have selection logic code.
+- How do you build more flexible menus with custom items? Dropdown relies on items inside of it to be selectable.
+
+For these reasons, we are going ahead with Menu component which will be for more flexible, clickable menus that are not selectable.
+
+## Is this a breaking change for Dropdown consumers?
+
+**No**. Everything that works currently will continue to work. We will deprecate the DropdownButton, DropdownLink usage and recommend moving to Menu for those usecase but not break the usage of DropdownButton or DropdownLink.
+
+How migration from deprecated API to recommended API will look like-
```

Comment:
Should we provide a codemod for this migration?

Reply:
We're not breaking it right now. If we decide to remove the API, we can provide a codemod then